pkg/sbproxy/notifications/producer_test.go

```diff
@@ -219,7 +315,102 @@ var _ = Describe("Notifications", func() {
 			})
 		})
 
-		Context("When websocket is connected", func() {
+		Context("When a server that accept mTLS exists", func() {
+			var tlsServer *httptest.Server
+			var tlsSocketServer *wsServer
+			BeforeEach(func() {
+				tlsServer, tlsSocketServer = newServerWithTLS()
+				producerSettings = &notifications.ProducerSettings{
+					MinPingPeriod:        100 * time.Millisecond,
+					ReconnectDelay:       100 * time.Millisecond,
+					PongTimeout:          20 * time.Millisecond,
+					ResyncPeriod:         300 * time.Millisecond,
+					PingPeriodPercentage: 60,
+					MessagesQueueSize:    10,
+				}
+			})
+
+			AfterEach(func() {
+				tlsServer.Close()
+			})
+
+			When("valid mTLS certificate and private keys are used", func() {
+				BeforeEach(func() {
+					smSettings = &sm.Settings{
```

Comment:
This can be extracted to a common BeforeEach, and in here only set the cert properties.
Same goes for the other test.

Reply:
We use different TLS settings in every each, is the suggestion to use common settings and modify only the certificate details in each?